File: backend/routes/payroll.py
from flask import Blueprint, request, jsonify
from utils.payrollCalculator import calculate_payroll

payroll_bp = Blueprint("payroll_bp", __name__)

@payroll_bp.route("/", methods=["POST"])
def run_payroll():
    data = request.json
    result = calculate_payroll(data["basic"], data.get("allowances",0), data.get("deductions",0))
    return jsonify({"message":"Payroll calculated","result":result})

File: backend/routes/payroll.js
import express from "express";
import { calculatePayroll } from "../utils/payrollCalculator.js";
const router = express.Router();

router.post("/", (req, res) => {
  const { basic, allowances, deductions } = req.body;
  const result = calculatePayroll(basic, allowances, deductions);
  res.json({ message: "Payroll calculated", result });
});

export default router;
